source/drivers/MicroBitCompassCalibrator.cpp

```diff
@@ -67,25 +242,27 @@ void MicroBitCompassCalibrator::calibrate(MicroBitEvent)
         uint8_t on;
     };
 
-    const int PERIMETER_POINTS = 12;
+    const int PERIMETER_POINTS = 25;
+
     const int PIXEL1_THRESHOLD = 200;
-    const int PIXEL2_THRESHOLD = 800;
+    const int PIXEL2_THRESHOLD = 680;
 
     wait_ms(100);
 
-	Matrix4 X(PERIMETER_POINTS, 4);
-    Point perimeter[PERIMETER_POINTS] = {{1,0,0}, {2,0,0}, {3,0,0}, {4,1,0}, {4,2,0}, {4,3,0}, {3,4,0}, {2,4,0}, {1,4,0}, {0,3,0}, {0,2,0}, {0,1,0}};
+    Point perimeter[PERIMETER_POINTS] = {{0,0,0}, {1,0,0}, {2,0,0}, {3,0,0}, {4,0,0}, {0,1,0}, {1,1,0}, {2,1,0}, {3,1,0}, {4,1,0}, {0,2,0}, {1,2,0}, {2,2,0}, {3,2,0}, {4,2,0}, {0,3,0}, {1,3,0}, {2,3,0}, {3,3,0}, {4,3,0}, {0,4,0}, {1,4,0}, {2,4,0}, {3,4,0}, {4,4,0}};
```

Comment:
Should these be static const so they don't take up RAM/stack?

Reply:
Yep, that makes sense. @finneyj in the original implementation did you have any reason for the way it was originally? Or intention to modify these points?